src/modules/flow/iio/nodes.c

```diff
@@ -101,15 +211,15 @@ create_device_channels(struct iio_direction_vector_data *mdata, int device_id)
 {
     struct sol_iio_channel_config channel_config = SOL_IIO_CHANNEL_CONFIG_INIT;
 
-    mdata->device = sol_iio_open(device_id, &mdata->config);
-    SOL_NULL_CHECK(mdata->device, false);
+    mdata->iio_base.device = sol_iio_open(device_id, &mdata->iio_base.config);
+    SOL_NULL_CHECK(mdata->iio_base.device, false);
 
 #define ADD_CHANNEL(_axis) \
```

Comment:
Not for this patch, but maybe we could find a way to unify this `ADD_CHANNEL` stuff... don't you think? But no need to worry about it now, it's another future improvement =D